casper/src/main/scala/coop/rchain/casper/merging/BlockIndex.scala

```diff
@@ -0,0 +1,70 @@
+package coop.rchain.casper.merging
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import coop.rchain.casper.protocol.ProcessedDeploy
+import coop.rchain.casper.util.EventConverter
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.block.StateHash.StateHash
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.rspace.history.HistoryRepository
+import coop.rchain.rspace.merger._
+import coop.rchain.rspace.syntax._
+import coop.rchain.rspace.trace.Produce
+
+import scala.collection.concurrent.TrieMap
+
+final case class BlockIndex(blockHash: BlockHash, deployChains: Vector[DeployChainIndex])
```

Comment:
Just a picky comment. Since the `blockHash` is not used anywhere. We can keep it from `Block` concept and leave it independent. 
I'd like to keep the block concept independent from virtual machine.

Reply:
For now, it is ok to leave it.

Reply:
@zsluedem MessageIndex?

Reply:
What I mean is not about the name. The `deployChains` is enough. BlockHash can be removed.

we can leave it as it is right now. No need to change.